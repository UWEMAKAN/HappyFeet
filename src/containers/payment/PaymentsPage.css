.PaymentsPage {
  font-size: 1.6rem;
}

.PaymentsPage h2 {
  font-size: 3rem;
  font-weight: 600;
  color: #2c9cee;
  margin-bottom: 4rem;
}
