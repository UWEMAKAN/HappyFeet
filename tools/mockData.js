module.exports = {
  shoes: [
    {
      id: 1,
      title: 'Blue Snickers',
      brand: 'Addidas',
      image: './photos/addidas-blue-1-600x600.jpg',
      color: 'Blue',
      price: 325,
      discount: 20,
      rating: 4.4,
    },
    {
      id: 2,
      title: 'Black Snickers',
      brand: 'Others',
      image: './photos/others-black-1-600x600.jpg',
      color: 'Black',
      price: 295,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 3,
      title: 'Black Snickers',
      brand: 'Others',
      image: './photos/others-black-2-600x600.jpg',
      color: 'Black',
      price: 325,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 4,
      title: 'White Snickers',
      brand: 'Nike',
      image: './photos/nike-white-1-600x600.jpg',
      color: 'White',
      price: 450,
      discount: 20,
      rating: 4.6,
    },
    {
      id: 5,
      title: 'Snickers',
      brand: 'Addidas',
      image: './photos/addidas-green-1-600x600.jpg',
      color: 'Green',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 6,
      title: 'High Hills',
      brand: 'Others',
      image: './photos/others-black-hills-1-600x600.jpg',
      color: 'Black',
      price: 375,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 7,
      title: 'White Snickers',
      brand: 'Nike',
      image: './photos/nike-white-2-600x600.jpg',
      color: 'White',
      price: 450,
      discount: 20,
      rating: 4.6,
    },
    {
      id: 8,
      title: 'Blue Snickers',
      brand: 'Addidas',
      image: './photos/addidas-blue-2-600x600.jpg',
      color: 'Blue',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 9,
      title: 'Black Leather Shoes',
      brand: 'Others',
      image: './photos/others-black-leather-1-600x600.jpg',
      color: 'Black',
      price: 425,
      discount: 25,
      rating: 4.3,
    },
    {
      id: 10,
      title: 'White Snickers',
      brand: 'Addidas',
      image: './photos/addidas-white-1-600x600.jpeg',
      color: 'White',
      price: 400,
      discount: 25,
      rating: 4.5,
    },
    {
      id: 11,
      title: 'Blue Snickers',
      brand: 'Others',
      image: './photos/others-blue-1-600x600.jpeg',
      color: 'Blue',
      price: 420,
      discount: 25,
      rating: 4.3,
    },
  ]
};
