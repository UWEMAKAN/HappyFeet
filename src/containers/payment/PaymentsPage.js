import React from 'react';
import propTypes from 'prop-types';
import Button from '../../components/Common/Button/Button';
import MyShoeCard from '../../components/MyShoeCard/MyShoeCard';
import NoResultsFound from '../../components/Common/NoResultsFound/NoResultsFound';
import './PaymentsPage.css';
import CreditCard from '../../components/CreditCard/CreditCard';

const PaymentsPage = ({ shoe, addToMyShoes }) => {
  const styles = {
    padding: '1rem 3rem'
  };

  return (
    <>
      {shoe.id
        ? (
          <div className="PaymentsPage">
            <h2>Make Payments</h2>
            <div>
              <MyShoeCard
                key={shoe.id}
                image={shoe.image}
                title={shoe.title}
                color={shoe.color}
                seller={shoe.seller}
                price={shoe.price}
              />
              <CreditCard />
              <Button name={`${shoe.id}`} link="/myshoes" clickHandler={() => addToMyShoes(shoe)} styles={styles} text="Pay" />
            </div>
          </div>
        )
        : <NoResultsFound message="Nothing To Buy" />}
    </>
  );
};

PaymentsPage.propTypes = {
  shoe: propTypes.instanceOf(Object).isRequired,
  addToMyShoes: propTypes.func.isRequired
};

export default PaymentsPage;
