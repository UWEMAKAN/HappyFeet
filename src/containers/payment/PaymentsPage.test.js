import React from 'react';
import { shallow } from 'enzyme';
import PaymentsPage from './PaymentsPage';
import { shoes } from '../../../tools/mockData';

const renderPaymentsPage = (args) => {
  const defaultProps = {
    shoe: {},
    addToMyShoes: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<PaymentsPage {...props} />);
};

describe('PaymentsPage Component', () => {
  test('should render PaymentsPage Component', () => {
    const shoe = shoes[0];
    const addToMyShoes = jest.fn();
    const wrapper = renderPaymentsPage({ shoe, addToMyShoes });
    expect.assertions(3);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').text()).toEqual('Make Payments');
    expect(wrapper.getElements()[0].props.children.props.children.length).toBe(2);
  });

  test('should render NoResults Found when there is no shoe to render', () => {
    const wrapper = renderPaymentsPage();
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });
});
