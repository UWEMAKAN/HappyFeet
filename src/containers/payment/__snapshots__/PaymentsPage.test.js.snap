// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentsPage Component should render NoResults Found when there is no shoe to render 1`] = `ShallowWrapper {}`;

exports[`PaymentsPage Component should render PaymentsPage Component 1`] = `ShallowWrapper {}`;
